### PR TITLE
feat (feature flags/ frontend) : Manage the feature flag endpoint and add a FF for the new resource status enhancements.

### PR DESCRIPTION
**Pull request by Med1233**
## Description

manage the feature flag endpoint on the frontend side in order to use the atom in different context

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by abuathier — DISMISSED** [review]

**Review by dmyios — DISMISSED** [review]

File: centreon/www/front_src/src/Resources/Listing/ResourceListing.cypress.spec.tsx [hunk 1]
```diff
@@ -115,6 +117,19 @@ const retrievedListingWithCriticalResources = {
   result: entitiesWithCriticalResources
 };
 
+const getPlatformFeatures = ({
+  enableThreeView = true
+}: {
+  enableThreeView?: boolean;
```
Comment (bdauria) [inline centreon/www/front_src/src/Resources/Listing/ResourceListing.cypress.spec.tsx:123]: shouldn't this be named treeView?

File: centreon/www/front_src/src/Main/atoms/platformFeaturesAtom.ts [hunk 2]
```diff
@@ -1,5 +1,12 @@
 import { atom } from 'jotai';
 
-import { PlatformFeatures } from '../../api/models';
+import { FeatureFlags, PlatformFeatures } from '../../api/models';
 
 export const platformFeaturesAtom = atom<PlatformFeatures | null>(null);
+export const FeatureFlagsAtom = atom<FeatureFlags | null>(
```
Comment (Thebarda) [inline centreon/www/front_src/src/Main/atoms/platformFeaturesAtom.ts:6]:
```suggestion
export const featureFlagsDerivedAtom = atom<FeatureFlags | null>(
```

File: centreon/www/front_src/src/Resources/Listing/index.tsx [hunk 3]
```diff
@@ -86,6 +87,8 @@ const ResourceListing = (): JSX.Element => {
   const panelWidth = useAtomValue(panelWidthStorageAtom);
   const forcedCheckInlineEndpoint = useAtomValue(forcedCheckInlineEndpointAtom);
   const visualization = useAtomValue(selectedVisualizationAtom);
+  const featureFlags = useAtomValue(FeatureFlagsAtom);
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Listing/index.tsx:90]:
```suggestion
  const featureFlags = useAtomValue(featureFlagsDerivedAtom);
```

File: centreon/www/front_src/src/Resources/Listing/index.tsx [hunk 4]
```diff
@@ -43,6 +43,7 @@ import {
   labelStatus,
   labelForcedCheckCommandSent
 } from '../translatedLabels';
+import { FeatureFlagsAtom } from '../../Main/atoms/platformFeaturesAtom';
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Listing/index.tsx:46]:
```suggestion
import { featureFlagsDerivedAtom } from '../../Main/atoms/platformFeaturesAtom';
```

**Review by abuathier — DISMISSED** [review]

**Review by Thebarda — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**